QModManager/API/SMLHelper/Crafting/ModCraftTreeRoot.cs

```diff
@@ -0,0 +1,110 @@
+﻿namespace QModManager.API.SMLHelper.Crafting
+{
+    using System.Linq;
+    using Patchers;
+    using UnityEngine.Assertions;
+
+    /// <summary>
+    /// The root node of a CraftTree. The whole tree starts here.
+    /// </summary>    
+    /// <seealso cref="ModCraftTreeLinkingNode" />
+    public class ModCraftTreeRoot : ModCraftTreeLinkingNode
+    {
+        private readonly string _schemeAsString;
+        private readonly CraftTree.Type _scheme;
+
+        internal override string SchemeAsString => _schemeAsString;
+        internal override CraftTree.Type Scheme => _scheme;
+
+        internal ModCraftTreeRoot(CraftTree.Type scheme, string schemeAsString)
+            : base("Root", TreeAction.None, TechType.None)
+        {
+            Assert.IsTrue((int)scheme > CraftTreeTypePatcher.startingIndex, "Custom CraftTree types must have an index higher than the in-game types.");
+
+            _schemeAsString = schemeAsString;
+            _scheme = scheme;
+            HasCustomTrees = true;
+        }
+
+        /// <summary>
+        /// Dynamically creates the CraftTree object for this crafting tree.
+        /// The CraftNode objects were created and linked as the classes of the ModCraftTreeFamily were created and linked.
+        /// </summary>
+        internal CraftTree CraftTree => new CraftTree(_schemeAsString, CraftNode);
+
+        /// <summary>
+        /// Populates a new ModCraftTreeRoot from a CraftNode tree.
+        /// </summary>
+        /// <param name="tree">The tree to create the ModCraftTreeRoot from.</param>
+        /// <param name="root"></param>
+        internal static void CreateFromExistingTree(CraftNode tree, ref ModCraftTreeLinkingNode root)
+        {
+            foreach (CraftNode node in tree)
+            {
+                if (node.action == TreeAction.Expand)
+                {
+                    ModCraftTreeTab tab = root.AddTabNode(node.id);
+                    ModCraftTreeLinkingNode thing = (ModCraftTreeLinkingNode)tab;
+                    CreateFromExistingTree(node, ref thing);
+                }
+
+                if (node.action == TreeAction.Craft)
+                {
+                    TechType techType = TechType.None;
+                    TechTypeExtensions.FromString(node.id, out techType, false);
+
+                    if (node.id == "SeamothHullModule2") techType = TechType.VehicleHullModule2;
+                    else if (node.id == "SeamothHullModule3") techType = TechType.VehicleHullModule3;
```

Comment:
Why is this hard-coding here?

Reply:
Looks like this is here to map the old, unused TechTypes for the Seamoth depth modules with the ones now in use.  
The Cyclops also has some old, unused Depth Upgrade TechTypes.

We could probably remove this entirely.